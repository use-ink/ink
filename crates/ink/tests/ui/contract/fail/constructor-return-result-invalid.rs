#![allow(unexpected_cfgs)]

#[ink::contract]
mod contract {
    #[ink(storage)]
    pub struct Contract {}

    #[derive(Debug, PartialEq, Eq)]
    #[ink::scale_derive(Encode, Decode, TypeInfo)]
    pub enum Error {
        Foo,
    }

    impl Contract {
        #[ink(constructor)]
        pub fn constructor() -> Result<u8, Error> {
            Ok(5_u8)
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
