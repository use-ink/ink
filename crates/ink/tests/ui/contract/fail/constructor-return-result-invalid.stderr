error[E0277]: the trait bound `ConstructorOutputValue<Result<u8, contract::Error>>: ConstructorOutput<Contract>` is not satisfied
  --> tests/ui/contract/fail/constructor-return-result-invalid.rs:14:9
   |
14 | /         pub fn constructor() -> Result<u8, Error> {
15 | |             Ok(5_u8)
16 | |         }
   | |_________^ the trait `ConstructorOutput<Contract>` is not implemented for `ConstructorOutputValue<Result<u8, contract::Error>>`
   |
   = help: the following other types implement trait `ConstructorOutput<C>`:
             ConstructorOutputValue<C>
             ConstructorOutputValue<Result<C, E>>

error[E0277]: the trait bound `Result<u8, contract::Error>: ConstructorReturnType<ContractRef>` is not satisfied
  --> tests/ui/contract/fail/constructor-return-result-invalid.rs:14:33
   |
14 |           pub fn constructor() -> Result<u8, Error> {
   |           -                       ^^^^^^^^^^^^^^^^^ the trait `ConstructorReturnType<ContractRef>` is not implemented for `Result<u8, contract::Error>`
   |  _________|
   | |
15 | |             Ok(5_u8)
16 | |         }
   | |_________- required by a bound introduced by this call
   |
   = help: the following other types implement trait `ConstructorReturnType<C>`:
             `Result<C, E>` implements `ConstructorReturnType<C>`
             `Result<Contract, E>` implements `ConstructorReturnType<ContractRef>`
note: required by a bound in `CreateBuilder::<E, ContractRef, CodeHash, Limits, Endowment, Args, Salt, Unset<ReturnType<()>>>::returns`
  --> $WORKSPACE/crates/env/src/call/create_builder.rs
   |
   |     pub fn returns<R>(
   |            ------- required by a bound in this associated function
...
   |         R: ConstructorReturnType<ContractRef>,
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `CreateBuilder::<E, ContractRef, CodeHash, Limits, Endowment, Args, Salt, Unset<ReturnType<()>>>::returns`

error[E0277]: the trait bound `ConstructorOutputValue<Result<u8, contract::Error>>: ConstructorOutput<Contract>` is not satisfied
 --> tests/ui/contract/fail/constructor-return-result-invalid.rs:4:16
  |
4 |     pub struct Contract {}
  |                ^^^^^^^^ the trait `ConstructorOutput<Contract>` is not implemented for `ConstructorOutputValue<Result<u8, contract::Error>>`
  |
  = help: the following other types implement trait `ConstructorOutput<C>`:
            ConstructorOutputValue<C>
            ConstructorOutputValue<Result<C, E>>
